### ISSUE TITLE
Infinite loop on autocorrect of Style/YodaCondition: Reverse the order of the operands

### ISSUE THREAD

**Issue by jaredhm** [bbatsov/rubocop#4426]
--------

## Expected behavior

When running rubocop as a rake task (with autocorrect) against a source file with snippet `yield(transition) == true`, I expect rubocop to fix the incorrect order of operands

## Actual behavior

cop `Style/YodaCondition` is triggered, the order of operands is auto-corrected (giving `yield(transition) == true`), the same cop is triggered again (this time for the corrected code), and then the stack gets blown as rubocop tries infinitely to correct itself. 

## Steps to reproduce the problem

I was able to reproduce in a simple project with the following:
```
ruby
└── whatever
    ├── Rakefile
    └── foo.rb
```

Rakefile:

```
require 'rubocop/rake_task'

RuboCop::RakeTask.new(:rubocop) do |task|
  task.options = ['-d', '-a']
  task.patterns = ['./*.rb']
end
```

foo.rb:

```
def some_fun(&_block)
  false || true == yield(42)
end

puts some_fun(&:is_even?)
```

Running `rake rubocop` from `ruby/whatever` produced the following stacktrace:

```
Running RuboCop...
For /Users/fmcbarson/Documents/Projects/ruby/whatever: configuration from /Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/config/default.yml
Inheriting configuration from /Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/config/enabled.yml
Inheriting configuration from /Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/config/disabled.yml
Inspecting 1 file
Scanning /Users/fmcbarson/Documents/Projects/ruby/whatever/foo.rb
C

Offenses:

foo.rb:2:11: C: [Corrected] Style/YodaCondition: Reverse the order of the operands true == yield(42).
  true || true == yield(42)
          ^^^^^^^^^^^^^^^^^
foo.rb:2:11: C: [Corrected] Style/YodaCondition: Reverse the order of the operands yield(42) == true.
  true || yield(42) == true
          ^^^^^^^^^^^^^^^^^

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in /Users/fmcbarson/Documents/Projects/ruby/whatever/foo.rb.
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:248:in `check_for_infinite_loop'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:231:in `block in iterate_until_no_changes'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:230:in `loop'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:230:in `iterate_until_no_changes'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:201:in `do_inspection_loop'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:111:in `block in file_offenses'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:121:in `file_offense_cache'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:109:in `file_offenses'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:100:in `process_file'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:78:in `block in each_inspected_file'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:75:in `each'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:75:in `reduce'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:75:in `each_inspected_file'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:67:in `inspect_files'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/runner.rb:39:in `run'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/cli.rb:82:in `execute_runner'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/cli.rb:28:in `run'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/rake_task.rb:48:in `run_cli'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/rake_task.rb:36:in `run_main_task'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/rake_task.rb:28:in `block (2 levels) in initialize'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/file_utils_ext.rb:58:in `verbose'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest/gems/rubocop-0.49.0/lib/rubocop/rake_task.rb:26:in `block in initialize'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/task.rb:250:in `block in execute'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/task.rb:250:in `each'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/task.rb:250:in `execute'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/task.rb:194:in `block in invoke_with_call_chain'
/Users/fmcbarson/.rvm/rubies/ruby-2.4.1-latest/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/task.rb:187:in `invoke_with_call_chain'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/task.rb:180:in `invoke'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:152:in `invoke_task'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:108:in `block (2 levels) in top_level'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:108:in `each'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:108:in `block in top_level'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:117:in `run_with_threads'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:102:in `top_level'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:80:in `block in run'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/lib/rake/application.rb:77:in `run'
/Users/fmcbarson/.rvm/gems/ruby-2.4.1-latest@global/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/Users/fmcbarson/.rvm/rubies/ruby-2.4.1-latest/bin/rake:22:in `load'
/Users/fmcbarson/.rvm/rubies/ruby-2.4.1-latest/bin/rake:22:in `<main>'
RuboCop failed!
```

Hopefully this is helpful!

## RuboCop version
```
-> rubocop -v
0.49.0
```

**Comment by danielbergamin** [comment]
Same experience with the following line, if it helps:

if __FILE__ != <filename as string literal>

**Comment by smakagon** [comment]
Thanks for report guys. I've managed to reproduce that on my local machine. 
Seems like these cases been fixed by: https://github.com/bbatsov/rubocop/commit/5700720cb92521e3a673e91ba517cc7884822826
I've pulled last master and that infinite loop didn't appear for described cases.
